src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -960,25 +1005,38 @@ public override Task<ImmutableArray<Project>> DetermineProjectsToSearchAsync(ISy
             Document document,
             SemanticModel semanticModel,
             Func<SyntaxToken, bool> tokensMatch,
-            Func<SyntaxToken, SyntaxNode> findParentNode,
+            Func<SyntaxToken, SyntaxNode>? findParentNode,
             CancellationToken cancellationToken)
         {
+            var findInGlobalSuppressions = ShouldFindReferencesInGlobalSuppressions(symbol, out var getDocumentationCommentId);
             var symbolsMatch = GetStandardSymbolsMatchFunction(symbol, findParentNode, document.Project.Solution, cancellationToken);
-            return FindReferencesInDocumentAsync(document, semanticModel, tokensMatch, symbolsMatch, cancellationToken);
+            return FindReferencesInDocumentAsync(document, semanticModel, tokensMatch,
+                symbolsMatch, getDocumentationCommentId, findInGlobalSuppressions, cancellationToken);
         }
 
         protected static async Task<ImmutableArray<FinderLocation>> FindReferencesInDocumentAsync(
             Document document,
             SemanticModel semanticModel,
             Func<SyntaxToken, bool> tokensMatch,
             Func<SyntaxToken, SemanticModel, (bool matched, CandidateReason reason)> symbolsMatch,
+            Func<string>? getDocumentationCommentId,
+            bool findInGlobalSuppressions,
             CancellationToken cancellationToken)
         {
-            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
 
             // Now that we have Doc Comments in place, We are searching for References in the Trivia as well by setting descendIntoTrivia: true
             var tokens = root.DescendantTokens(descendIntoTrivia: true);
-            return FindReferencesInTokens(document, semanticModel, tokens, tokensMatch, symbolsMatch, cancellationToken);
+            var references = FindReferencesInTokens(document, semanticModel, tokens, tokensMatch, symbolsMatch, cancellationToken);
+
+            if (!findInGlobalSuppressions)
+                return references;
+
+            RoslynDebug.Assert(getDocumentationCommentId != null);
+            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var referencesInGlobalSuppressions = await FindReferencesInDocumentInsideGlobalSuppressionsAsync(
+                document, semanticModel, syntaxFacts, getDocumentationCommentId, cancellationToken).ConfigureAwait(false);
+            return references.AddRange(referencesInGlobalSuppressions);
```

Comment:
something feels duplicative here since i saw this exact code above.  not sure what's possible to clean this up.  maybe a core helper that both existing methods calls?

Reply:
Yeah, unfortunately I wasn't able to pull this into common code as the containing methods operate on different token groups.